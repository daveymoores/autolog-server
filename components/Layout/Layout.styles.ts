import css from "styled-jsx/css";

export default css`
  .page-layout {
    padding: 25px;
  }

  @media screen and (min-width: 900px) {
    .page-layout {
      padding: 50px;
    }
  }

  main {
    padding: 25px 0;
  }

  @media screen and (min-width: 900px) {
    main {
      padding: 50px 0;
    }
  }
`;
