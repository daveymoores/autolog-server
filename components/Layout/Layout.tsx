import React from "react";

import Footer from "../Footer/Footer";
import Navigation from "../Navigation/Navigation";

const Layout: React.FC<React.PropsWithChildren> = (props) => {
  return (
    <div className="page-layout">
      <Navigation />
      <main>{props.children}</main>
      <Footer />
    </div>
  );
};

export default Layout;
