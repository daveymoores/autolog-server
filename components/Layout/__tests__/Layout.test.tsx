import React from "react";
import renderer from "react-test-renderer";

import Layout from "../Layout";

describe("Layout", () => {
  it("renders Layout unchanged", () => {
    const tree = renderer.create(<Layout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
