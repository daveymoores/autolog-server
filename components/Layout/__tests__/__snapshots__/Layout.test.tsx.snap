// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders Layout unchanged 1`] = `
<div
  className="page-layout"
>
  <div
    className="container grid grid-cols-12 mt-8 lg:mt-16 mx-auto lg:max-w-screen-xl font-semibold"
  >
    <div
      className="col-span-12 md:col-span-10 md:col-start-2 lg:col-span-8 lg:col-start-3"
    >
      <nav
        className="flex justify-between align-middle"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": 15,
                "margin": 0,
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": 114,
              }
            }
          >
            <img
              data-nimg="fixed"
              decoding="async"
              src="/_next/image?url=%2Flogo.svg&w=256&q=75"
              srcSet="/_next/image?url=%2Flogo.svg&w=128&q=75 1x, /_next/image?url=%2Flogo.svg&w=256&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </span>
        </a>
        <a
          href="/documentation"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Documentation
        </a>
      </nav>
    </div>
  </div>
  <main />
  <footer
    className="bg-black mt-24"
  >
    <nav
      className="container font-semibold flex justify-center flex-col align-middle items-center gap-2 py-16"
    >
      <a
        href="https://github.com/daveymoores/autolog"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Github
      </a>
      <a
        href="https://www.linkedin.com/in/daveymoores"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Linkedin
      </a>
      <div
        className="mt-24"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": 15,
                "margin": 0,
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": 114,
              }
            }
          >
            <img
              data-nimg="fixed"
              decoding="async"
              src="/_next/image?url=%2Flogo.svg&w=256&q=75"
              srcSet="/_next/image?url=%2Flogo.svg&w=128&q=75 1x, /_next/image?url=%2Flogo.svg&w=256&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </span>
        </a>
      </div>
    </nav>
  </footer>
</div>
`;
